WeatherApp/DataSource/MainWeatherViewDataSources/HourlyWeatherCollectionDataSource.swift

```diff
@@ -17,7 +17,7 @@ final class HourlyWeatherCollectionDataSource: NSObject, UICollectionViewDataSou
             let cellModels = cellModels,
             let cellModel = cellModels[.hourlyWeatherCollectionCellModel] as? HourlyWeatherCollectionCellModel
         else { return 0 }
-        return cellModel.weatherModel.hourly.count
+        return cellModel.dataModel.count
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
```

Comment:
guard cellModels.count > indexPath.item else { return nil }
let cellModel = cellModels[indexPath.item]
let cell = collectionView.dequeueReusableCell(withReuseIdentifier: cellModel.cellIdentifier, for: indexPath)
if let fillableCell = cell as? FillableCell {
fillableCell.fill(for: cellModel)
} 
return cell